beacon-chain/operations/attestations/kv/aggregated.go

```diff
@@ -88,6 +97,58 @@ func (c *AttCaches) aggregateUnaggregatedAttestations(ctx context.Context, unagg
 	return nil
 }
 
+// aggregateParallel aggregates attestations in parallel for `atts` and saves them in the pool,
+// returns the unaggregated attestations that weren't able to aggregate.
+// Given CPU cores, it creates a channel of size `n` and spawns `n` goroutines to aggregate attestations
+func (c *AttCaches) aggregateParallel(atts map[[32]byte][]*ethpb.Attestation, leftOver map[[32]byte]bool) map[[32]byte]bool {
+	var leftoverLock sync.Mutex
+	wg := sync.WaitGroup{}
+
+	n := runtime.GOMAXPROCS(0) // defaults to the value of runtime.NumCPU
+	ch := make(chan []*ethpb.Attestation, n)
```

Comment:
This is probably not worth optimizing, but you can write `math.Min(n, len(atts))`